DashboardClient/OpcUaTypeReader.cpp

```diff
@@ -12,392 +12,330 @@
 #include <regex>
 #include <Exceptions/OpcUaException.hpp>
 
- namespace Umati
-{
-    namespace Dashboard
-    {
-
-        OpcUaTypeReader::OpcUaTypeReader(
-            std::shared_ptr<IDashboardDataClient> pIClient,
-            std::vector<std::string> expectedObjectTypeNamespaces,
-            std::vector<Umati::Util::NamespaceInformation> namespaceInformations)
-            : m_expectedObjectTypeNamespaces(std::move(expectedObjectTypeNamespaces)),
-              m_pClient(pIClient)
-        {
-            for (auto const &el: namespaceInformations) {
-                m_availableObjectTypeNamespaces[el.Namespace] = el;
-                for (auto const &e: el.Types) {
-                    m_identificationTypeOfTypeDefinition.insert(std::make_pair(e, el.IdentificationType));
-                    m_knownMachineTypeDefinitions.emplace_back(e);
-                    m_subTypeDefinitionToKnownMachineTypeDefinition.insert(std::make_pair(e,e));
-                }
+namespace Umati {
+namespace Dashboard {
+
+OpcUaTypeReader::OpcUaTypeReader(
+  std::shared_ptr<IDashboardDataClient> pIClient,
+  std::vector<std::string> expectedObjectTypeNamespaces,
+  std::vector<Umati::Util::NamespaceInformation> namespaceInformations)
+  : m_expectedObjectTypeNamespaces(std::move(expectedObjectTypeNamespaces)), m_pClient(pIClient) {
+  for (auto const &el : namespaceInformations) {
+    m_availableObjectTypeNamespaces[el.Namespace] = el;
+    for (auto const &e : el.Types) {
+      m_identificationTypeOfTypeDefinition.insert(std::make_pair(e, el.IdentificationType));
+      m_knownMachineTypeDefinitions.emplace_back(e);
+      m_subTypeDefinitionToKnownMachineTypeDefinition.insert(std::make_pair(e, e));
+    }
+  }
+}
+
+OpcUaTypeReader::~OpcUaTypeReader() {
+  for (auto &entry : *m_typeMap.get()) {
+    entry.second.get()->SpecifiedChildNodes.get()->clear();
+  }
+}
+
+void OpcUaTypeReader::readTypeDictionaries() {
+  m_pClient->readTypeDictionaries();
+  m_pClient->buildCustomDataTypes();
+  m_pClient->updateCustomTypes();
+}
+
+void OpcUaTypeReader::readTypes() {
+  std::vector<std::string> notFoundObjectTypeNamespaces;
+  auto bidirectionalTypeMap = std::make_shared<std::map<ModelOpcUa::NodeId_t, std::shared_ptr<ModelOpcUa::StructureBiNode>>>();
+  initialize(notFoundObjectTypeNamespaces);
+  LOG(INFO) << "Browsing variable types.";
+  this->BaseDataVariableTypes.push_back(NodeId_BaseVariableType);
+  this->BaseDataVariableTypes.push_back(ModelOpcUa::NodeId_t({"", "i=63"}));
```

Comment:
Whe are the NodeIds with the ns0-Uri requred, when the one without?
May is be enought to use only the one without the namespaceUri?